lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -553,6 +553,153 @@ static void addImplicitDistributedActorStoredProperties(ClassDecl *decl) {
+
+  auto *funcName = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Function, loc, /*Implicit=*/true);
+  funcName->setType(staticStringType);
+  funcName->setBuiltinInitializer(staticStringInit);
+
+  auto *file = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::FileID, loc, /*Implicit=*/true);
+  file->setType(staticStringType);
+  file->setBuiltinInitializer(staticStringInit);
+
+  auto *line = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Line, loc, /*Implicit=*/true);
+  line->setType(uintType);
+  line->setBuiltinInitializer(uintInit);
+
+  auto *column = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Column, loc, /*Implicit=*/true);
+  column->setType(uintType);
+  column->setBuiltinInitializer(uintInit);
+
+  auto *call = CallExpr::createImplicit(
+      ctx, ref, { className, funcName, file, line, column }, {});
+  call->setType(ctx.getNeverType());
+  call->setThrows(false);
+
+  SmallVector<ASTNode, 2> stmts;
+  stmts.push_back(call); // something() -> Never
+  // stmts.push_back(new (ctx) ReturnStmt(SourceLoc(), /*Result=*/nullptr)); // FIXME: this causes 'different types for return type: String vs. ()'
+  auto body = BraceStmt::create(ctx, SourceLoc(), stmts, SourceLoc(),
+      /*implicit=*/true);
+
+
+
+  return { body, /*isTypeChecked=*/true };
+}
+
+static Identifier makeRemoteFuncIdentifier(FuncDecl* func) {
+  auto &C = func->getASTContext();
+  auto localFuncName = func->getBaseIdentifier().str().str();
+  auto remoteFuncIdent = C.getIdentifier("_remote_" + localFuncName);
+  return remoteFuncIdent;
+}
+
+/// Create a remote stub for the passed in \c func.
+/// The remote stub function is not user accessible and mirrors the API of
+/// the local function. It is always throwing, async, and user-inaccessible.
+///
+/// ```
+/// // func greet(name: String) { ... }
+/// dynamic <access> func _remote_greet(name: String) async throws {
+///     fatalError(...)
+/// }
+/// ```
+///
+/// and is intended to be replaced by a transport library by providing an
+/// appropriate @_dynamicReplacement function.
+static void addImplicitRemoteActorFunction(ClassDecl *decl, FuncDecl *func) {
+  auto &C = decl->getASTContext();
+  auto parentDC = decl;
+
+  auto remoteFuncIdent = makeRemoteFuncIdentifier(func);
+
+  auto params = func->getParameters();
+  auto genericParams = func->getGenericParams();
+  auto resultTy = func->getResultInterfaceType();
+
+  DeclName name(C, remoteFuncIdent, params);
+  auto *const remoteFuncDecl = FuncDecl::createImplicit(
+      C, StaticSpellingKind::None, name, /*NameLoc=*/SourceLoc(),
+      /*Async=*/true, /*Throws=*/true,
+      /*GenericParams=*/genericParams, params,
+      resultTy, parentDC);
+
+  // *dynamic* because we'll be replacing it with specific transports
+  remoteFuncDecl->getAttrs().add(
+      new (C) DynamicAttr(/*implicit=*/true));
+
+  // @_distributedActorIndependent
```

Comment:
(this will be renamed _distributedActorNonisolated, it is not an user visible attribute)